components/core/tools/scripts/lib_install/macos/install-all.sh

```diff
@@ -7,9 +7,15 @@ set -e
 set -u
 
 brew update
+
+# Install CMake v3.31.6 since we're using yaml-cpp that doesn't yet support CMake v4 yet
+cmake_formula_path=/tmp/cmake.rb
+curl --fail --show-error --location --output "${cmake_formula_path}" \
+  https://raw.githubusercontent.com/Homebrew/homebrew-core/b4e46db74e74a8c1650b38b1da222284ce1ec5ce/Formula/c/cmake.rb
```

Comment:
```suggestion
curl --fail --location --show-error \
  https://raw.githubusercontent.com/Homebrew/homebrew-core/b4e46db74e74a8c1650b38b1da222284ce1ec5ce/Formula/c/cmake.rb \
  --output "${cmake_formula_path}"
```

Reply:
The version of curl on macOS seems to require that the options come before the URL.

Reply:
Sorry, scratch that. I was misinterpreting the error I saw.

Reply:
hmm... the suggested command seems to work on my mac:
```
~ % curl --version
curl 8.7.1 (x86_64-apple-darwin24.0) libcurl/8.7.1 (SecureTransport) LibreSSL/3.3.6 zlib/1.2.12 nghttp2/1.63.0
Release-Date: 2024-03-27
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM SPNEGO SSL threadsafe UnixSockets
```

maybe macos-13 is using an older version?